consensus/core/src/tests/base_committer_declarative_tests.rs

```diff
@@ -365,3 +365,102 @@ async fn indirect_skip() {
         panic!("Expected a skipped leader")
     };
 }
+
+/// Here we setup a situation where theres is an undecided leader (D)
+/// but later we encounter B which votes for D to make it committed directly
+#[tokio::test]
+async fn test_equivocating_direct_commit() {
+    telemetry_subscribers::init_for_testing();
+    let context = Arc::new(Context::new_for_test(4).0);
+    let dag_state = Arc::new(RwLock::new(DagState::new(
+        context.clone(),
+        Arc::new(MemStore::new()),
+    )));
+    let committer = BaseCommitterBuilder::new(context.clone(), dag_state.clone()).build();
+
+    let dag_str = "DAG {
+        Round 0 : { 4 },
+        Round 1 : { * },
+        Round 2 : { * },
+        Round 3 : { * }
+        Round 4 : { 
+            A -> [],
+            B -> [],
+            C -> [*],
+            D -> [*],
+        },
+        Round 5 : { * },
```

Comment:
You can remove building Round 5 here as you will build it below and it will achieve the same result. IIUC I don't believe your goal is to equivocate all blocks for round 5, you just want to equivocate for block B4

Reply:
this makes sense indeed